packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx

```diff
@@ -0,0 +1,63 @@
+import { Button, Icon } from '@wordpress/components';
+import { lock } from '@wordpress/icons';
+import clsx from 'clsx';
+import { useTranslate } from 'i18n-calypso';
+
+interface DomainsTableSSLCellProps {
+	domainManagementLink: string;
+	sslStatus: 'active' | 'pending' | 'disabled' | null;
+}
+
+export default function DomainsTableSSLCell( {
+	domainManagementLink,
+	sslStatus,
+}: DomainsTableSSLCellProps ) {
+	const translate = useTranslate();
+
+	const getSSLStatusText = () => {
+		if ( sslStatus === 'active' ) {
+			return translate( 'Active' );
+		}
+		if ( sslStatus === 'pending' ) {
+			return translate( 'Pending' );
+		}
+		if ( sslStatus === 'disabled' ) {
+			return translate( 'Disabled' );
+		}
+	};
+
+	const handleClick = ( event: React.MouseEvent< HTMLButtonElement, MouseEvent > ) => {
+		event.stopPropagation();
+		window.location.href = `${ domainManagementLink }?ssl-open=true`;
+	};
+
+	return (
+		<div className="domains-table-row__ssl-cell">
+			{ sslStatus && (
+				<Icon
+					className={ clsx( 'domains-table-row__ssl-icon', {
+						[ 'domains-table-row__ssl-icon__active' ]: sslStatus === 'active',
+						[ 'domains-table-row__ssl-icon__pending' ]: sslStatus === 'pending',
+						[ 'domains-table-row__ssl-icon__disabled' ]: sslStatus === 'disabled',
+					} ) }
+					icon={ lock }
+				/>
+			) }
+			{ sslStatus !== null ? (
+				<Button
+					className={ clsx( 'domains-table-row__ssl-status-button', {
+						[ 'is-active' ]: sslStatus === 'active',
+						[ 'is-pending' ]: sslStatus === 'pending',
+						[ 'is-disabled' ]: sslStatus === 'disabled',
+					} ) }
+					variant="link"
+					onClick={ handleClick }
```

Comment:
```suggestion
					href={ `${ domainManagementLink }?ssl-open=true` }
```
Can we pass the `href` directly and make it a true link?

Reply:
Technically yes, but I don't think it's ideal.

Our cell's parent `DomainsTableRow` has its own click handler, which is what opens `domains/manage` for the various cells. If we want our cell to behave differently and add a url parameter, we need a separate click handler that can stop propagation, at which point the new click handler also has to handle navigating us to the URL with the added param.

All of which means that while passing an `href` _will_ render the button as a true link, the `href` we've passed will effectively be ignored in favor of the click handler. We'd have to put the URL in both places (click handler and `href`), but only one of them would actually control navigation, which I worry is going to be unpleasant for future devs looking at this code.

...That is of course unless there's away around all of that I'm not thinking of yet.

Reply:
Good point 👍 That's what we do [here](https://github.com/Automattic/wp-calypso/blob/b81d5e143a5c5110841cba6d7beedaea6a160fb5/packages/domains-table/src/domains-table/domains-table-email-indicator.tsx#L65-L71) and [here](https://github.com/Automattic/wp-calypso/blob/89e0ede9a17cc29fd843ac277cf0cc12fcd72e47/packages/domains-table/src/domains-table/domains-table-row.tsx#L130-L136), though, so I think it's reasonable to use that same logic in this case.

Reply:
Making this into a regular anchor element instead of a `<Button>` component would also reduce the spacing between the icon and the button, which is currently a bit excessive (I compared it with the design doc).

Reply:
Oh yes, that's much better. Passing the `href` to the `Button` would render an anchor but introduce the problems I mentioned. Just using an `anchor` with an `onClick` to stop propagation avoids that issue entirely, thanks!